pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1327,23 +1332,37 @@ public static Address getToAddressId(@NonNull EasyPostUtil easyPostUtil, KitRequ
                 if (kitRequestSettings.getHasCareOF() != 1) {
                     // aside form singular, all other studies should go here and proceed with normal label
                     toAddress = easyPostUtil.createAddress(participant, kitRequestSettings.getPhone());
-                    return toAddress;
+                } else {
+                    toAddress = getAddressForStudiesWithCareOfField(easyPostUtil, kitRequestSettings, participant, ddpInstanceDto);
                 }
-                toAddress = getAddressForStudiesWithCareOfField(easyPostUtil, kitRequestSettings, participant, ddpInstanceDto);
             }
         }
+        if (toAddress == null) {
+            throw new DsmInternalError("Couldn't get address for kit request");
```

Comment:
Add a bit more detail about the kit request in question--anything you can include in the exception message about method params?